src/ServiceControlInstaller.Engine/Instances/BaseService.cs

```diff
@@ -61,21 +61,27 @@ public bool TryStopService()
             try
             {
                 Service.WaitForStatus(ServiceControllerStatus.Stopped, TimeSpan.FromSeconds(60));
-                var t = new Task(() =>
-                {
-                    while (!HasUnderlyingProcessExited())
-                    {
-                        Thread.Sleep(100);
-                    }
-                });
-                t.Wait(5000);
```

Comment:
@fourpastmidnight The other changes look ok but would it be easier to just change this line to `return t.Wait(5000);`?